rest_api/test/test_rest_api.py

```diff
@@ -208,226 +123,307 @@ def populated_client(client: TestClient):
+        assert 200 == response.status_code
+        # Ensure `run` was called with the expected parameters
+        mocked_pipeline.run.assert_called_with(query=q, params=params, debug=False)
+
+
+def test_query_with_filter_list(client):
+    q = "Who made the PDF specification?"
+    params = {"TestRetriever": {"filters": {"test_key": ["test_value", "another_value"]}}}
+    with mock.patch("rest_api.controller.search.query_pipeline") as mocked_pipeline:
+        # `run` must return a dictionary containing a `query` key
+        mocked_pipeline.run.return_value = {"query": q}
+        response = client.post(url="/query", json={"query": q, "params": params})
+        assert 200 == response.status_code
+        # Ensure `run` was called with the expected parameters
+        mocked_pipeline.run.assert_called_with(query=q, params=params, debug=False)
+
+
+def test_query_with_invalid_filter(client):
+    q = "Who made the PDF specification?"
+    request_params = {"TestRetriever": {"filters": {"test_key": "invalid_value"}}}
+    expected_params = {"TestRetriever": {"filters": {"test_key": ["invalid_value"]}}}
+    with mock.patch("rest_api.controller.search.query_pipeline") as mocked_pipeline:
+        # `run` must return a dictionary containing a `query` key
+        mocked_pipeline.run.return_value = {"query": q}
+        response = client.post(url="/query", json={"query": q, "params": request_params})
+        assert 200 == response.status_code
+        # Ensure `run` was called with the expected parameters. In this case,
+        # `_format_filters` will fix the `filters` format within the params
+        mocked_pipeline.run.assert_called_with(query=q, params=expected_params, debug=False)
+
+
+def test_query_with_no_documents_and_no_answers(client):
+    q = "Who made the PDF specification?"
+    with mock.patch("rest_api.controller.search.query_pipeline") as mocked_pipeline:
+        # `run` must return a dictionary containing a `query` key
+        mocked_pipeline.run.return_value = {"query": q}
+        response = client.post(url="/query", json={"query": q})
+        assert 200 == response.status_code
+        response_json = response.json()
+        assert response_json["documents"] == []
+        assert response_json["answers"] == []
 
 
-def test_write_feedback_without_id(populated_client: TestClient, api_document_store: BaseDocumentStore):
-    feedback = deepcopy(FEEDBACK)
+def test_write_feedback(client, feedback):
+    response = client.post(url="/feedback", json=feedback)
+    assert 200 == response.status_code
+    # Ensure `write_labels` was called on the Document Store instance passing a list
+    # containing only one label
+    args, _ = MockDocumentStore.mocker.write_labels.call_args
+    labels = args[0]
+    assert len(labels) == 1
+    # Ensure all the items that were in `feedback` are part of the stored label
+    label = labels[0].to_dict()
+    for k, v in feedback.items():
+        assert label[k] == v
```

Comment:
Isn't it better to do simply:

```python
# Ensure all the items that were in `feedback` are part of the stored label
assert labels[0].to_dict() == feedback
```

Or `label` has more keys than what `feedback` contained?